Quadriga/src/main/java/edu/asu/spring/quadriga/service/dictionary/mapper/impl/DictionaryShallowMapper.java

```diff
@@ -45,6 +45,21 @@
 	}
 	
 	@Override
+    @Transactional
+    public List<IDictionary> getNonAssociatedProjectDictionaries(String projectId) throws QuadrigaStorageException {
+	    List<DictionaryDTO> dictionaryDTOList = dictDao.getNonAssociatedProjectDictionaries(projectId);
+        
+        List<IDictionary> dictionaryList = new ArrayList<IDictionary>();
+        if(dictionaryDTOList != null) {
+            for(DictionaryDTO dictionaryDTO: dictionaryDTOList) {
+                dictionaryList.add(createDictionaryProxy(dictionaryDTO));
+            }
+        }
+        
+        return dictionaryList;

```

Comment:
try to resuse existing methods. You can iterate over all non-associated projects in the manager and just call the mapper to create the proxies. mappers should only map DTOs to domain objects, they shouldn't recreate all retrieval logic.